### PR TITLE
TPM2: Introduce NVUndefineSpaceEx 

### PR DESCRIPTION
**Pull request by ChriMarMe**
Add NVUndefineSpaceEx
Remove encodeUndefineSpace
Rework NVUndefineSpace to work on top of NVUndefineSpace

Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: tpm2/tpm2.go [hunk 1]
```diff
@@ -1164,25 +1164,24 @@ func NVIncrement(rw io.ReadWriter, handle tpmutil.Handle, authString string) err
...
-
-// NVUndefineSpace removes an index from TPM's NV storage.
-func NVUndefineSpace(rw io.ReadWriter, ownerAuth string, owner, index tpmutil.Handle) error {
-	Cmd, err := encodeUndefineSpace(ownerAuth, owner, index)
+	auth, err := encodeAuthArea(authArea)
 	if err != nil {
 		return err
 	}
-	_, err = runCommand(rw, TagSessions, CmdUndefineSpace, tpmutil.RawBytes(Cmd))
+	cmd, err := concat(out, auth)
```
Comment (twitchy-jsonp) [inline tpm2/tpm2.go:1183]: Check the error from `concat()`:

```go
if err != nil {
	return err
}
```

File: tpm2/tpm2.go [hunk 2]
```diff
@@ -1164,25 +1164,24 @@ func NVIncrement(rw io.ReadWriter, handle tpmutil.Handle, authString string) err
...
-	if err != nil {
-		return nil, err
-	}
+// NVUndefineSpace removes an index from TPM's NV storage.
+func NVUndefineSpace(rw io.ReadWriter, ownerAuth string, owner, index tpmutil.Handle) error {
+	authArea := AuthCommand{Session: HandlePasswordSession, Attributes: AttrContinueSession, Auth: []byte(ownerAuth)}
+	return NVUndefineSpaceEx(rw, owner, index, authArea)
+}
+
+// NVUndefineSpaceEx gets AuthCommand struct for more flexible use
```
Comment (twitchy-jsonp) [inline tpm2/tpm2.go:1173]: Use full sentences for doc comments. Perhaps:
```go
// NVUndefineSpaceEx removes an index from NVRAM. Unlike, NVUndefineSpace(), custom command
// authorization can be provided.
```

Comment (ChriMarMe) [reply tpm2/tpm2.go]: did that.

File: tpm2/tpm2.go (hunk 1 above)
Comment (ChriMarMe) [reply tpm2/tpm2.go:1184]: done

**Comment by ChriMarMe** [comment]
Anything else?

**Comment by ChriMarMe** [comment]
ping

**Review by twitchy-jsonp — APPROVED** [review]

**Pull request merged**